check.js

```diff
@@ -0,0 +1,459 @@
+/**
+ * Contract checks, assertions, and verifications
+ * @module check
+ */
+
+const constitution = require('./constitution');
+const ships = require('./ships');
+const polls = require('./polls');
+const pool = require('./pool');
+const utils = require('./utils');
+const reasons = require('./resources/reasons.json');
+
+const MAXGALAXY = 255;
+const MAXSTAR = 65535;
+const MAXPLANET = 4294967295;
+
+/**
+ * Check if something is a ship.
+ * @param {Number} ship - Ship token.
+ * @return {Bool} True if a ship, false otherwise.
+ */
+function isShip(ship) {
+  return (typeof ship === 'number' && ship >= 0 && ship <= MAXPLANET);
+}
+
+/**
+ * Check if something is a galaxy.
+ * @param {Number} ship - Ship token.
+ * @return {Bool} True if a galaxy, false otherwise.
+ */
+function isGalaxy(ship) {
+  return (typeof ship === 'number' && ship >= 0 && ship <= MAXGALAXY);
+}
+
+/**
+ * Check if something is a star.
+ * @param {Number} ship - Ship token.
+ * @return {Bool} True if a star, false otherwise.
+ */
+function isStar(ship) {
+  return (typeof ship === 'number' && ship > MAXGALAXY && ship <= MAXSTAR);
+}
+
+/**
+ * Check if something is a planet.
+ * @param {Number} ship - Ship token.
+ * @return {Bool} True if a planet, false otherwise.
+ */
+function isPlanet(ship) {
+  return (typeof ship === 'number' && ship > MAXSTAR && ship <= MAXPLANET);
+}
+
+/**
+ * Check if a ship is a parent of another ship.
+ * @param {Number} ship - Ship token.
+ * @return {Bool} True if a parent, false otherwise.
+ */
+function isParent(ship) {
+  return (typeof ship === 'number' && ship > 0 && ship <= MAXSTAR);
+}
+
+/**
+ * Check if a ship is a child of another ship.
+ * @param {Number} ship - Ship token.
+ * @return {Bool} True if a child, false otherwise.
+ */
+function isChild(ship) {
+  return (typeof ship === 'number' && ship > MAXGALAXY && ship <= MAXPLANET);
+}
+
+/**
+ * Check if a poll is active.
+ * @param {Object} poll - A poll object.
+ * @return {Bool} True if active, false otherwise.
+ */
+function pollIsActive(poll) {
+  let now = Math.round(new Date().getTime()/1000);
+  let end = poll.start + poll.duration;
+  return now < end;
+}
+
+/**
+ * Check if a poll can be started.
+ * @param {Object} poll - A poll object.
+ * @return {Bool} True if so, false otherwise.
+ */
+function canStartPoll(poll) {
+  let now   = Math.round(new Date().getTime()/1000);
+  let start = poll.start + poll.duration + poll.coolDown;
+  return now > start;
+}
+
+/**
+ * Check if a ship has an owner.
+ * @param {Object} contracts - An Urbit contracts object.
+ * @param {Number} tokenId - Ship token.
+ * @return {Promise<Bool>} True if so, false otherwise.
+ */
+async function hasOwner(contracts, ship) {
+  if (typeof ship === 'object') {
```

Comment:
This (with typo fixed) actually breaks all the assertion functions, since they're also expected to work on promises (so, the last check doesn't necessarily hold).  It took me a little bit of time to figure out WTF was going on.

I think the existing check is ok for now, since the aim of this code is to expose a generic API, rather than prevent bogus inputs *per se*.

That said, branching on types generally gives me the willies, no matter what language I'm in, so I'm inclined to export an alternative pure ships API where passing an ship object (rather than integer) is done explicitly.  One could require `cjs.ships` or `cjs.ships.pure` separately, for example, depending on what one is interested in doing.  But I think figuring out the details of that is something for down the road.